aesm-client/tests/live_quote.rs

```diff
@@ -41,3 +44,61 @@ fn live_quote() {
         )
         .expect("quote result");
 }
+
+pub fn get_algorithm_id(key_id : &Vec<u8>) -> u32 {
+    const ALGORITHM_OFFSET : usize = 154;
+
+    let mut bytes: [u8; 4] = Default::default();
+    bytes.copy_from_slice(&key_id[ALGORITHM_OFFSET..ALGORITHM_OFFSET+4]);
+    u32::from_le_bytes(bytes)
+}
+
+#[cfg(not(windows))]
+#[test]
+fn test_ecdsa_quote() {
+    const NONCE: [u8; 16] = [0; 16];
+    const SGX_QL_ALG_ECDSA_P256 : u32 = 2;
+
+    let mut device = IsgxDevice::new()
+        .unwrap()
+        .einittoken_provider(AesmClient::new())
+        .build();
+
+    let client = AesmClient::new();
+
+    let key_ids = client.get_supported_att_key_ids().unwrap();
+
+    // Print the keys for debugging
+    for i in key_ids.iter() {
+        let ptr = i.as_ptr() as *const sgx_att_key_id_ext_t;
+        unsafe {
+            println!("att_key_type: {:?}; spid: {:?}; base: id {:?}; version: {:?}; mrsigner_length {:?}; mrsigner {:?}; prod_id {:?}; extended_prod_id {:?}; config_id {:?}; family_id {:?}; algorithm_id {:?}",
+                     (*ptr).att_key_type, (*ptr).spid, (*ptr).base.id, (*ptr).base.version, (*ptr).base.mrsigner_length,
+                     (*ptr).base.mrsigner.split_at(16), (*ptr).base.prod_id, (*ptr).base.extended_prod_id, (*ptr).base.config_id.split_at(16), (*ptr).base.family_id, (*ptr).base.algorithm_id);
+        }
+    }
+
+    // Select the ECDSA key that will be used later
+    let ecdsa_key_id = key_ids.into_iter().find(|id| SGX_QL_ALG_ECDSA_P256 == get_algorithm_id(id));
+
+    if let Some(key) = ecdsa_key_id {
+        match client.init_quote_ex(key.clone()) {
+            Err(Error::AesmCode(AesmError::UnexpectedError_1)) => {
```

Comment:
I don't think this case should be handled separately. This test shouldn't pass if it's not supported. Why not just do `client.init_quote_ex(key.clone()).unwrap()`?

Reply:
added a comment in case someone sees the failure and wants to debug, hopefully CI still passes.